de.metas.payment.esr/src/main/java/de/metas/payment/esr/dataimporter/impl/camt54/ESRDataImporterCamt54.java

```diff
@@ -124,294 +100,111 @@ public ESRDataImporterCamt54(@NonNull final I_ESR_Import header, @NonNull final
 	 * 
-	 * @param stmtBuilder builder to which the individual {@link ESRTransaction}s are added.
-	 * @param ctrlQty
-	 * @param ntry
-	 * @return the given {@code ctrlQty}, plus the <code>NbOfTxs</code> of the given {@code ntry}'s {@code ntryDtl}s (if any).
+	 * @param fileName
+	 * @return
 	 */
-	@VisibleForTesting
-	BigDecimal iterateEntryDetails(
-			@NonNull final ESRStatementBuilder stmtBuilder,
-			@Nullable final BigDecimal ctrlQty,
-			@NonNull final ReportEntry8 ntry)
+	private boolean isVersion2Schema(@NonNull final String namespaceURI)
 	{
-		BigDecimal newCtrlQty = ctrlQty;
-		for (final EntryDetails7 ntryDtl : ntry.getNtryDtls())
-		{
-			if (ctrlQty.compareTo(CTRL_QTY_AT_LEAST_ONE_NULL) == 0
-					|| ntryDtl.getBtch() == null || ntryDtl.getBtch().getNbOfTxs() == null)
-			{
-				// the current ntryDtl has no control qty, or an earlier one already didn't have a control qty
-				newCtrlQty = CTRL_QTY_AT_LEAST_ONE_NULL;
-			}
-			else
-			{
-				final BigDecimal augend = new BigDecimal(ntryDtl.getBtch().getNbOfTxs());
-				if (ctrlQty.compareTo(CTRL_QTY_NOT_YET_SET) == 0)
-				{
-					// not yet set
-					newCtrlQty = augend;
-				}
-				else
-				{
-					newCtrlQty = newCtrlQty.add(augend);
-				}
-			}
-
-			final List<ESRTransaction> transactions = iterateTransactionDetails(ntry, ntryDtl);
-			stmtBuilder.transactions(transactions);
-
-		} // ntryDtl
-
-		return newCtrlQty;
-	}
-
-	private List<ESRTransaction> iterateTransactionDetails(
-			@NonNull final ReportEntry8 ntry,
-			@NonNull final EntryDetails7 ntryDtl)
-	{
-		final List<ESRTransaction> transactions = new ArrayList<>();
-
-		for (final EntryTransaction8 txDtl : ntryDtl.getTxDtls())
-		{
-			final ESRTransactionBuilder trxBuilder = ESRTransaction.builder();
-
-			new ReferenceStringHelper().extractAndSetEsrReference(txDtl, trxBuilder);
-
-			verifyTransactionCurrency(txDtl, trxBuilder);
-
-			extractAmountAndType(ntry, txDtl, trxBuilder);
-
-			final ESRTransaction esrTransaction = trxBuilder
-					.accountingDate(asTimestamp(ntry.getBookgDt()))
-					.paymentDate(asTimestamp(ntry.getValDt()))
-					.esrParticipantNo(ntry.getNtryRef())
-					.transactionKey(mkTrxKey(txDtl))
-					.build();
-			transactions.add(esrTransaction);
-		}
-		return transactions;
+		return Objects.equal("urn:iso:std:iso:20022:tech:xsd:camt.054.001.02", namespaceURI);
 	}
-
-	private void extractAmountAndType(
-			@NonNull final ReportEntry8 ntry,
-			@NonNull final EntryTransaction8 txDtls,
-			@NonNull final ESRTransactionBuilder trxBuilder)
+	
+	private String getNameSpaceURI(@NonNull final XMLStreamReader reader)
 	{
-		// credit-or-debit indicator
-
-		final ActiveOrHistoricCurrencyAndAmount transactionDetailAmt = txDtls.getAmt();
-
-		final BigDecimal amount = transactionDetailAmt.getValue()
-				.multiply(getCrdDbtMultiplier(txDtls.getCdtDbtInd()))
-				.multiply(getRvslMultiplier(ntry));
-		trxBuilder.amount(amount);
-
-		if (txDtls.getCdtDbtInd() == CreditDebitCode.CRDT)
+		try
```

Comment:
OK..I'm not 100% sure, if this this try-catch makes sense..
This method is already called from a similar try-catch block, so why did you choose to have this one too?

Reply:
Because doesn't; compile otherwise. There is a XMLStreamException

Reply:
in that case, couldn't you add a `throws XMLStreamException` to the method declaration?

Reply:
Right! Got it!